rcl/src/rcl/logging_rosout.c

```diff
@@ -236,10 +237,31 @@ void rcl_logging_rosout_output_handler(
 {
   rosout_map_entry_t entry;
   rcl_ret_t status = RCL_RET_OK;
+  char * name_dup = NULL;
+  rcutils_allocator_t allocator = rcutils_get_default_allocator();
   if (!__is_initialized) {
     return;
   }
   rcutils_ret_t rcutils_ret = rcutils_hash_map_get(&__logger_map, &name, &entry);
+  // if logger name not found, continue to check if its parent logger name exists
+  if (RCUTILS_RET_OK != rcutils_ret) {
+    name_dup = rcutils_strdup(name, allocator);
+    if (NULL == name_dup) {
+      RCUTILS_SAFE_FWRITE_TO_STDERR("unable to allocate memory\n");
+      return;
+    }
+    do {
+      char * name_sep = strrchr(name_dup, RCUTILS_LOGGING_SEPARATOR_STRING[0]);
+      if (NULL == name_sep) {
+        break;
+      }
+      name_dup[(size_t)(name_sep - name_dup)] = '\0';
+      if (!rcutils_logging_logger_is_enabled_for(name_dup, severity)) {
```

Comment:
This kind of seems like the wrong layer of the logging to do this in.  Everywhere else, this is done at the `rcutils` layer, while here we are introducing this at the `rcl` layer.